src/asmcnc/core_UI/hoverable.py

```diff
@@ -0,0 +1,500 @@
+        elif keycode == 'i':
+            self.inspect_widget()
+        if keycode == 'h':
+            self.print_help()
+        elif keycode == 'w':
+            self.open_close_component_popup()
+        elif keycode == 'e':
+            self.switch_edit_mode()
+        elif keycode == 'm':
+            self.switch_multi_select_state()
+        elif keycode == 't':
+            self.switch_snap_mode()
+        elif 273 <= key <= 276:  # check for arrow keys to move the widget
+            if self.snap_mode:
+                self.snap_widget(key)
+            else:
+                self.move_widget(key)
+        elif key in [43, 270]:  # '+' keys
+            self.adjust_step_width(1)
+        elif key in [45, 269]:  # '-' keys
+            self.adjust_step_width(-1)
+        elif key == 127:  # del key
+            self.delete_widget()
+        elif key in [303, 304]:  # shift
+            self.add_remove_widget_to_selection()
+
+        # Return True to accept the key. Otherwise, it will be used by the system.
+        return True
+
+    def switch_snap_mode(self):
+        self.snap_mode = not self.snap_mode
+        Logger.debug('snap_mode: {}'.format(self.snap_mode))
+    def switch_lock_state(self):
+        if not self.locked:
+            Logger.debug('LOCKED to: {}'.format(self.get_widget_name_class(self.widget)))
+            self.locked = True
+        else:
+            Logger.debug('UNLOCKED')
+            self.locked = False
+
+    def open_close_component_popup(self):
+        self.dispatch('on_show_component_popup')
+
+    def adjust_step_width(self, value):
+        self.step_width += value
+        Logger.debug('Step width: {}'.format(self.step_width))
+
+    def switch_multi_select_state(self):
+        """Switches multiselect mode between True and False."""
+        self.multiselect = not self.multiselect
+        Logger.debug('multiselect: {}'.format(self.multiselect))
+
+    def switch_edit_mode(self):
+        """Switches edit_mode between True and False."""
+        self.edit_mode = not self.edit_mode
+        Logger.debug('edit_mode: {}'.format(self.edit_mode))
+
+    def delete_widget(self):
+        """Deletes the selected widget from the screen."""
+        if self.widget:
+            self.widget.parent.remove_widget(self.widget)
+            self.widget = None
+            Logger.warning('Deleted selected widget!')
+
+    def snap_widget(self, key):
+        """
+        Snaps the selected widget to the closest other widget within the same container.
+        If no widget is in the way, the selected widget will snap to the border.
+
+        Uses default padding.
+        """
+        if not self.widget:
+            return
+        # get all siblings within the same container:
+        widgets_to_check = self.widget.parent.children
+        target_widget = None
+        min_distance = 9999  # used to determine the closes widget
+        w = self.widget   # just for readability
```

Comment:
Some places in the file use 'w' some use 'widget'.

Reply:
I think only this functions uses 'w'. It is just for readability so the comparisons 't.x < w.x < t.right ' are shorter.